util/stmtsummary/statement_summary.go

```diff
@@ -621,14 +633,15 @@ func newStmtSummaryByDigestElement(sei *StmtExecInfo, beginTime int64, intervalS
 		prevSQL: sei.PrevSQL,
 		// samplePlan needs to be decoded so it can't be truncated.
 		samplePlan:   sei.PlanGenerator(),
-		sampleUser:   sei.User,
 		indexNames:   sei.StmtCtx.IndexNames,
 		minLatency:   sei.TotalLatency,
 		firstSeen:    sei.StartTime,
 		lastSeen:     sei.StartTime,
 		backoffTypes: make(map[fmt.Stringer]int),
+		authUsers:    make(map[string]struct{}),
 	}
 	ssElement.add(sei, intervalSeconds)
+	ssElement.authUsers[sei.User] = struct{}{}
```

Comment:
This should be moved to `stmtSummaryByDigestElement::add`. `newStmtSummaryByDigestElement` is only recorded once, so only the first user is recorded, but `authUsers` should record all users.